core/src/main/java/org/elasticsearch/search/SearchModule.java

```diff
@@ -556,7 +569,7 @@ private void registerBuiltinQueryParsers() {
         registerQueryParser(new CommonTermsQueryParser(), CommonTermsQueryParser.QUERY_NAME_FIELD);
         registerQueryParser(new SpanMultiTermQueryParser(), SpanMultiTermQueryParser.QUERY_NAME_FIELD);
         QueryParser<FunctionScoreQueryBuilder> functionScoreParser = (QueryParseContext c) -> FunctionScoreQueryBuilder
-                .fromXContent((String name) -> functionScoreParsers.get(name), c);
+                .fromXContent(this::lookupScoreFunctionParser, c);
```

Comment:
I find this BiFunction business a bit hard to read. I prefer what we did for queries, which we have a specific IndicesQueriesRegistry class for. We used to have a similar ScoreFunctionMapper or something, maybe that should be brought back and called ScoreFunctionRegistry? I know it's one more class but I think it is more consistent with what we do with queries and easier to read. Thoughts?